pkg/daemon/daemon.go

```diff
@@ -352,46 +385,54 @@ func (d *daemon) AddPeriodicUpdate() {
 
 		// Get configured PKEY for network and add the relevant POD GUIDs as members of the PKey via Subnet Manager
 		if ibCniSpec.PKey != "" && len(guidList) != 0 {
-			pKey, err := utils.ParsePKey(ibCniSpec.PKey)
+			var pKey int
+			pKey, err = utils.ParsePKey(ibCniSpec.PKey)
 			if err != nil {
 				log.Error().Msgf("failed to parse PKey %s with error: %v", ibCniSpec.PKey, err)
 				continue
 			}
 
-			if err = d.smClient.AddGuidsToPKey(pKey, guidList); err != nil {
-				log.Error().Msgf("failed to config pKey with subnet manager %s with error: %v",
-					d.smClient.Name(), err)
+			// Try to add pKeys via subnet manager in backoff loop
+			if err = wait.ExponentialBackoff(backoffValues, func() (bool, error) {
+				if err = d.smClient.AddGuidsToPKey(pKey, guidList); err != nil {
+					log.Warn().Msgf("failed to config pKey with subnet manager %s with error : %v",
+						d.smClient.Name(), err)
+					return false, nil
+				}
+				return true, nil
+			}); err != nil {
+				log.Error().Msgf("failed to config pKey with subnet manager %s", d.smClient.Name())
 				continue
 			}
 		}
 
 		// Update annotations for PODs that finished the previous steps successfully
 		var removedGUIDList []net.HardwareAddr
 		for _, pi := range passedPods {
-			err := d.updatePodNetworkAnnotation(pi, &removedGUIDList)
-			if err != nil {
-				failedPods = append(failedPods, pi.pod)
-				log.Error().Msgf("%v", err)
-				continue
-			}
+			d.updatePodNetworkAnnotation(pi, &removedGUIDList)
```

Comment:
can we keep the original updatePodNetworkAnnotation signature and return error, then log it here ? it seems clearer to me that way so the the main flow is "aware" of the failure

Reply:
Ok, If U asking

Reply:
I prefer we have it as i suggested